src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -3164,7 +3164,9 @@ def aks_agentpool_add(cmd, client, resource_group_name, cluster_name, nodepool_n
     upgradeSettings = AgentPoolUpgradeSettings()
     taints_array = []
 
-    creationData = None
+    # load model CreationData
+    from azure.cli.command_modules.acs.decorator import AKSModels
+    CreationData= AKSModels(cmd, ResourceType.MGMT_CONTAINERSERVICE).CreationData
     if snapshot_id:
```

Comment:
maybe need 'creationData = None' here, a lot of test failures happen here.

Reply:
I think you are right. Otherwise, we don't have a default value for creationData.